components/dashboard/src/components/Modal.tsx

```diff
@@ -1,20 +1,47 @@
+import { Disposable, DisposableCollection } from "@gitpod/gitpod-protocol";
+
 export default function Modal(props: {
     children: React.ReactChild[] | React.ReactChild,
     visible: boolean,
     closeable?: boolean,
     className?: string,
-    onClose: () => void
+    onClose: () => void,
+    onEnter?: () => boolean
 }) {
+    const disposable = new DisposableCollection();
+    const close = () => {
+        disposable.dispose();
+        props.onClose();
+    }
     if (!props.visible) {
         return null;
     }
-    setTimeout(() => window.addEventListener('click', props.onClose, { once: true }), 0);
+    setTimeout(() => {
```

Comment:
I'd suggest to use `useEffect` hook instead in order to guarantee that the event listeners are received by the visible modals only. 

This is what I did in parallel.

```
     const handler = (e: MouseEvent) => {
         // ...
     }
    useEffect(() => {
        if (!props.visible) {
            window.removeEventListener("click", handler)
            return;
        }

        window.addEventListener("click", handler)

        // do something on unmount
        return () => {  };
    }, [props.visible]);
```

This hook's invocation depends on `props.visible` change, which gives us control of the event handlers.

OTOH, given the `setTimeout` solution, a `<Modal visible={false}></Modal>` would already register the event handler.